.travis.yml

```diff
@@ -6,22 +6,17 @@ stages:
     branch = develop OR 
     branch =~ /.*\.(?i:build)$/ OR 
     type = pull_request 
-- name: build images
+- name: build-images
   if: branch = main OR 
     branch = develop OR 
-    branch =~ /.*\.(?i:build)$/ OR 
-    type = pull_request OR 
-    branch =~ /^v\d+\.\d+\.\d+(-S*)?$/
-- name: assemble manifest
-  if: branch = main OR 
-    branch = develop OR 
-    branch =~ /.*\.(?i:build)$/ OR 
-    type = pull_request OR 
     branch =~ /^v\d+\.\d+\.\d+(-S*)?$/
 - name: build-docs
   if: branch = main OR 
     branch = develop OR 
     branch =~ /.*\.(?i:build)$/
```

Comment:
we keep the trigger on docs for `.build` branches but not the build images trigger? Also, I think it would be good if PRs trigger builds (and tests if we have them), but *not* a publish (publish on new pushes to the main/develop branch but not on the PR)

Reply:
The logic should now do tests for main, develop, tags, and PRs; image build/push for main, develop, and tags; and docs for main and develop.

Tests include building on mac, and linux with both gcc and clang.  In those three cases the unit tests are also run.

Building but not pushing the images as part of the tests is left as a future to-do item.